Samples/BackgroundBlur/BackgroundBlur/Capture.cpp

```diff
@@ -0,0 +1,517 @@
+                SetEvent(m_pManager->m_hEvent);
+            }
+            else
+            {
+                // This is an event we don't know about, we don't really care and there's
+                // no clean way to report the error so just set the event and fall through.
+                SetEvent(m_pManager->m_hEvent);
+            }
+        }
+
+        return S_OK;
+    }
+    else
+    {
+        pEvent->AddRef();  // The application will release the pointer when it handles the message.
+        PostMessage(m_hwnd, WM_APP_CAPTURE_EVENT, (WPARAM)pEvent, 0L);
+    }
+
+    return S_OK;
+}
+
+HRESULT CreateDX11Device(_Out_ ID3D11Device** ppDevice, _Out_ ID3D11DeviceContext** ppDeviceContext, _Out_ D3D_FEATURE_LEVEL* pFeatureLevel )
+{
+    HRESULT hr = S_OK;
+    static const D3D_FEATURE_LEVEL levels[] = {
+        D3D_FEATURE_LEVEL_11_1,
+        D3D_FEATURE_LEVEL_11_0,  
+        D3D_FEATURE_LEVEL_10_1,
+        D3D_FEATURE_LEVEL_10_0,
+        D3D_FEATURE_LEVEL_9_3,
+        D3D_FEATURE_LEVEL_9_2,
+        D3D_FEATURE_LEVEL_9_1
+    };
+
+    // Find the adapter with the most video memory
+    UINT i = 0;
+
+    std::vector <IDXGIAdapter*> vAdapters;
+    com_ptr<IDXGIAdapter> pBestAdapter;
+
+    com_ptr<IDXGIAdapter> spAdapter;
+    com_ptr<IDXGIFactory> spFactory;
+    DXGI_ADAPTER_DESC desc;
+    hr = CreateDXGIFactory1(IID_PPV_ARGS(spFactory.put()));
+    size_t maxVideoMem = 0;
+    while (spFactory->EnumAdapters(i, spAdapter.put()) != DXGI_ERROR_NOT_FOUND) 
+    {
+        spAdapter->GetDesc(&desc);
+        if (desc.DedicatedVideoMemory > maxVideoMem)
+        {
+            spAdapter.copy_to(pBestAdapter.put());
+            maxVideoMem = desc.DedicatedVideoMemory;
+        }
+        spAdapter = nullptr;
+        ++i;
+    }
+
+    hr = D3D11CreateDevice(
+        pBestAdapter.get(),
+        D3D_DRIVER_TYPE_UNKNOWN,
+        nullptr,
+        D3D11_CREATE_DEVICE_VIDEO_SUPPORT,
+        levels,
+        ARRAYSIZE(levels),
+        D3D11_SDK_VERSION,
+        ppDevice,
+        pFeatureLevel,
+        ppDeviceContext
+        );
+    
+    if(SUCCEEDED(hr))
+    {
+        com_ptr<ID3D10Multithread> pMultithread;
+        hr =  ((*ppDevice)->QueryInterface(IID_PPV_ARGS(pMultithread.put())));
+
+        if(SUCCEEDED(hr))
+        {
+            pMultithread->SetMultithreadProtected(TRUE);
+        }
+
+        
+    }
+
+    return hr;
+}
+
+HRESULT CreateD3DManager()
+{
+    HRESULT hr = S_OK;
+    D3D_FEATURE_LEVEL FeatureLevel;
+    com_ptr<ID3D11DeviceContext> pDX11DeviceContext;
+    
+    hr = CreateDX11Device(&g_pDX11Device, pDX11DeviceContext.put(), &FeatureLevel);
+
+    if(SUCCEEDED(hr))
+    {
+        hr = MFCreateDXGIDeviceManager(&g_ResetToken, &g_pDXGIMan);
+    }
+
+    if(SUCCEEDED(hr))
+    {
+        hr = g_pDXGIMan->ResetDevice(g_pDX11Device, g_ResetToken);
+    }
+        
+    return hr;
+}
+
+HRESULT
+CaptureManager::InitializeCaptureManager(HWND hwndPreview, IUnknown* pUnk)
```

Comment:
If this is a sample I would follow 

HRESULT PerformOperation(int operation) noexcept try
{
    vector<int> items = GetItems(operation);
    SomeFunction(items);
    return S_OK;
}
CATCH_RETURN();

instead of using on throw news, this would allow for modification in future without letting an exception leak or sub-function changing its impl

see [https://www.osgwiki.com/wiki/Cpp_Core_Guidelines#PATTERNS.1:_Follow_the_error_code_based_Try_pattern](https://www.osgwiki.com/wiki/Cpp_Core_Guidelines#PATTERNS.1:_Follow_the_error_code_based_Try_pattern)